localflavor/gr/forms.py

```diff
@@ -60,3 +61,56 @@ def clean(self, value):
         if mod != check:
             raise ValidationError(self.error_messages['invalid'])
         return val
+
+
+class GRSocialSecurityNumberCodeField(Field):
+    """
+    Greek social security number (AMKA) field.
+
+    The allow_test_value option can be used to enable the usage of the
+    non valid 00000000000 (11 zeros) value for testing and development
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a valid greek social security number (AMKA - 11 digits).'),
+    }
+
+    def __init__(self, allow_test_value=False, *args, **kwargs):
+        self.allow_test_value = allow_test_value
+        super(GRSocialSecurityNumberCodeField, self).__init__(*args, **kwargs)
+
+    def check_date(self, val):
+        try:
+            datetime.datetime.strptime(val[:6], '%d%m%y')
+        except:
+            raise ValidationError(self.error_messages['invalid'])
+
+    def check_lunh(self, val):
+        # Uses the Luhn algorithm: https://en.wikipedia.org/wiki/Luhn_algorithm
+        s = 0
+        for idx, d in enumerate(reversed(list(map(int, val)))):
+            if (idx + 1) % 2 == 0:
+                d *= 2
+            if d > 9:
+                d -= 9
+            s += d
+        if s % 10 != 0:
+            raise ValidationError(self.error_messages['invalid'])
+
+    def clean(self, value):
+        super(GRSocialSecurityNumberCodeField, self).clean(value)
+        if value in EMPTY_VALUES:
+            return ''
```

Comment:
I think you won't need this check if you inherit from `RegexField` since there are checks for empty values in `CharField`.

Reply:
Actually this:

```
        if value in self.empty_values:
            return self.empty_value

```
was needed or else the test ``test_forms_char_field_empty_value_allows_none`` would fail.

Reply:
Ok cool. Thanks for catching that.